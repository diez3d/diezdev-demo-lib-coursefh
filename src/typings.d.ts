declare module '*.css' {
    const content: { [className: string]: string };
    export default content;
}
// Add more modules if you want more extensions files
declare module "*.jpg" {
    const value: any;
    export default value;
}