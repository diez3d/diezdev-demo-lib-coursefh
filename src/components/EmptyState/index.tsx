import React from 'react'
import { FC } from 'react'

import './styles.css'


interface EmptyStateProps {
  classCustom?: string
  text?: string
  image?: {
    src: string
    description: string
  }
}

export const EmptyState: FC<EmptyStateProps> = ({ classCustom, text, image }) => {
  return (
    <div className={`emptyState ${classCustom}`}>
      {image && (
        <img
          src={image.src}
          alt={image.description}
          title={image.description}
        />
      )}
      {text && <span className="emptyState__text">{text}</span>}
    </div>
  )
}
