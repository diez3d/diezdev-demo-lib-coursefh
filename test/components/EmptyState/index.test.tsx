import React from 'react';
import renderer from 'react-test-renderer';
import {EmptyState} from '../../../src/components';
/* import * as ReactDOM from 'react-dom'; */

describe('EmptyState', () => {
  it('debe mostrar text como prop', () => {
    const wrapper = renderer.create(
        <EmptyState text="mi mock text prop"/>
    )
    expect(wrapper.toJSON()).toMatchSnapshot()

  });
});