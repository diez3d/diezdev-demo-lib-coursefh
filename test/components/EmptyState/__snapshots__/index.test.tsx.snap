// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyState debe mostrar text como prop 1`] = `
<div
  className="emptyState undefined"
>
  <span
    className="emptyState__text"
  >
    mi mock text prop
  </span>
</div>
`;
